sklearn/preprocessing/tests/test_knn_imputation.py

```diff
@@ -0,0 +1,220 @@
+from __future__ import division
+import numpy as np
+
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_array_almost_equal
+from sklearn.utils.testing import assert_raises
+from sklearn.utils.testing import assert_false
+
+from sklearn.preprocessing.knn_imputation import KNNImputer
+from sklearn.random_projection import sparse_random_matrix
+
+
+def test_knn_imputation_shape():
+    # Verify the shapes of the imputed matrix for different weights and
+    # number of neighbors.
+    n_rows = 10
+    n_cols = 2
+    X = np.random.rand(n_rows, n_cols)
+    X[0, 0] = np.nan
+
+    for weights in ['uniform', 'distance']:
+        for n_neighbors in range(1, 6):
+            imputer = KNNImputer(n_neighbors=n_neighbors, weights=weights)
+            X_imputed = imputer.fit_transform(X)
+            assert_equal(X_imputed.shape, (n_rows, n_cols))
+
+
+def test_knn_imputation_zero():
+    # Test imputation when missing_values == 0
```

Comment:
So turns out there is a bug in NearestNeighbors at the moment. The issue is that it does not pass missing_values to pairwise_distances so the latter obviously only considers NaN as missing value resulting in a different behavior when missing_values=0 compared to missing_values=NaN. 
I have not fixed it yet since you since you seemed ambivalent about supporting the option of passing missing_values itself. So I will only fix this if you would like to keep this feature. However if you think we should restrict missing to NaN then I won't fix the bug itself but make the other necessary changes. Please let me know what you prefer. Thanks.